src/main/kotlin/net/ccbluex/liquidbounce/features/command/commands/client/fakeplayer/CommandFakePlayer.kt

```diff
@@ -0,0 +1,314 @@
+                                BigDecimal(fakePlayer.z).setScale(1, RoundingMode.HALF_UP).toDouble()
+                            )
+                        )
+                    )
+                    anyRemoved = true
+                    return@removeIf true
+                } else {
+                    return@removeIf false
+                }
+            }
+
+            if (!anyRemoved) {
+                chat(warning((command.result("noFakePlayerNamed", name))))
+                chat(regular(command.result("currentlySpawned")))
+                fakePlayers!!.forEach { fakePlayer ->
+                    chat(regular("- " + fakePlayer.name.string))
+                }
+            }
+        }
+        .build()
+
+    private fun clearCommand() = CommandBuilder
+        .begin("clear")
+        .handler { _, _ ->
+            checkInGame()
+
+            if (fakePlayers == null || fakePlayers!!.isEmpty()) {
+                throw CommandException(translation("liquidbounce.command.fakeplayer.noFakePlayers"))
+            }
+
+            fakePlayers!!.removeIf { fakePlayer ->
+                world.removeEntity(fakePlayer.id, Entity.RemovalReason.DISCARDED)
+                return@removeIf true
+            }
+            fakePlayers = null
+        }
+        .build()
+
+    @Suppress("SpellCheckingInspection")
+    private fun startRecordingCommand() = CommandBuilder
+        .begin("startrecording")
+        .handler { command, _ ->
+            if (recording) {
+                throw CommandException(command.result("alreadyRecording"))
+            }
+
+            recording = true
+            snapshots = LinkedList<PosPoseSnapshot>()
+            chat(regular(command.result("startedRecording")))
+            notification(
+                "FakePlayer",
+                command.result("startedRecordingNotification"),
+                NotificationEvent.Severity.INFO
+            )
+        }
+        .build()
+
+    @Suppress("SpellCheckingInspection")
+    private fun endRecordingCommand() = CommandBuilder
+        .begin("endrecording")
+        .parameter(
+            ParameterBuilder
+                .begin<String>("name")
+                .verifiedBy(ParameterBuilder.STRING_VALIDATOR)
+                .optional()
+                .build()
+        )
+        .handler { command, args ->
+            if (!recording) {
+                throw CommandException(command.result("notRecording"))
+            }
+
+            if (snapshots!!.isEmpty()) {
+                throw CommandException(command.result("somethingWentWrong"))
+            }
+
+            spawn(args, true)
+            stopRecording()
+        }
+        .build()
+
+    /**
+     * Adds a new fake player.
+     *
+     * Note: a moving player requires [snapshots] to be not null.
+     *
+     * @param moving true if the fake player should play a recording.
+     */
+    private fun spawn(args: Array<Any>, moving: Boolean) {
+        checkInGame()
```

Comment:
This should be checked in the command handler since this function's job is to spawn the player.